vision_datasets/key_value_pair/operations.py

```diff
@@ -0,0 +1,32 @@
+from ..common import DatasetTypes, MultiImageCocoDictGenerator, \
+    MultiImageDatasetSingleTaskMerge, CocoDictGeneratorFactory, ManifestMergeStrategyFactory
+    
+from .manifest import KeyValuePairLabelManifest, KeyValuePairDatasetManifest
+
+_DATA_TYPE = DatasetTypes.KEY_VALUE_PAIR
+
+
+@CocoDictGeneratorFactory.register(_DATA_TYPE)
+class KeyValuePairCocoDictGenerator(MultiImageCocoDictGenerator):
+    def process_labels(self, coco_ann, label: KeyValuePairLabelManifest):
+        coco_ann[KeyValuePairLabelManifest.LABEL_KEY] = label.key_value_pairs
+        if label.text is not None:
+            coco_ann[KeyValuePairLabelManifest.INPUT_KEY] = label.text
+
+    def _generate_images(self, manifest: KeyValuePairDatasetManifest):
+        images = super()._generate_images(manifest)
+        # add metadata field if exists
+        for img, img_manifest in zip(images, manifest.images):
+            if img_manifest.additional_info is not None and 'metadata' in img_manifest.additional_info:
+                img['metadata'] = img_manifest.additional_info['metadata']
```

Comment:
what is the metadata used for? It seems not mentioned in the doc

Reply:
added to the doc, it is text attributes of interest that might be helpful for the task, e.g. size of a screw, location the image is taken.